### PR TITLE
Stop Audio & Video when user taps on share NFT button

### PR DESCRIPTION
**Pull request by ahmadrns**
When user click on share button while previewing NFT the Audio or video should stops accordingly.

### CODE REVIEW

File: wallet/lib/pages/detailed_asset_view/owner_view_view_model.dart [hunk 1]
```diff
@@ -396,7 +396,21 @@ class OwnerViewViewModel extends ChangeNotifier {
     notifyListeners();
   }
 
+  void pauseMedia() {
```
Comment (kjawadDeveloper2) [inline wallet/lib/pages/detailed_asset_view/owner_view_view_model.dart:399]: Can you add test for this?

**Review by kjawadDeveloper2 — CHANGES_REQUESTED** [review]

**Comment by kjawadDeveloper2** [comment]
There are issue with testing . We should not create video player or audio player. We should use mock repository from the mockito not the one from the mock folder. Using that you should stub values.

File: wallet/test/mocks/video_player.mocks.dart [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+  ) : super(
+          parent,
+          parentInvocation,
+        );
+}
+
+/// A class which mocks [VideoPlayerController].
+///
+/// See the documentation for Mockito's code generation for more information.
+class MockVideoPlayerController extends _i1.Mock
```
Comment (kjawadDeveloper2) [inline wallet/test/mocks/video_player.mocks.dart:40]: No need for this.

File: wallet/test/widget_testing/pages/owner_view_screen_test.dart [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+import '../../mocks/mock_audio_player.dart';
+import '../../mocks/mock_constants.dart';
+import '../../mocks/mock_repository.dart';
+import '../../mocks/mock_share_helper.dart';
+import '../../mocks/mock_video_player.dart';
+import '../../mocks/mock_wallet_store.dart';
+import '../../mocks/video_player.mocks.dart';
+import '../extension/size_extension.dart';
+
+void main() {
```
Comment (kjawadDeveloper2) [inline wallet/test/widget_testing/pages/owner_view_screen_test.dart:26]: @GeneratedMocks([OwnerViewModel]) only this is needed.

**Review by kjawadDeveloper2 — CHANGES_REQUESTED** [review]

**Comment by kjawadDeveloper2** [comment]
Kindly also add descriptions.

**Comment by ahmadrns** [comment]
> There are issue with testing . We should not create video player or audio player. We should use mock repository from the mockito not the one from the mock folder. Using that you should stub values.

Audio & Video have different helper class like `AudioPlayerHelper` & `videoPlayerHelper` which is responsible for playing & pausing media, how can we mock repository to stub the values of `play`,`pause` audio & video

**Comment by kjawadDeveloper2** [comment]
> > There are issue with testing . We should not create video player or audio player. We should use mock repository from the mockito not the one from the mock folder. Using that you should stub values.
> 
> Audio & Video have different helper class like `AudioPlayerHelper` & `videoPlayerHelper` which is responsible for playing & pausing media, how can we mock repository to stub the values of `play`,`pause` audio & video

Mock the viewModel not helpers.

**Comment by ahmadrns** [comment]
> > > There are issue with testing . We should not create video player or audio player. We should use mock repository from the mockito not the one from the mock folder. Using that you should stub values.
> > 
> > 
> > Audio & Video have different helper class like `AudioPlayerHelper` & `videoPlayerHelper` which is responsible for playing & pausing media, how can we mock repository to stub the values of `play`,`pause` audio & video
> 
> Mock the viewModel not helpers.

I have mocked the viewModel itself and not using helper classes in this test case but I didn't deleted it yet as when we have to write test cases for helper classes and we need them. So, can i delete it for now or leave it as it is?

**Comment by kjawadDeveloper2** [comment]
> > > > There are issue with testing . We should not create video player or audio player. We should use mock repository from the mockito not the one from the mock folder. Using that you should stub values.
> > > 
> > > 
> > > Audio & Video have different helper class like `AudioPlayerHelper` & `videoPlayerHelper` which is responsible for playing & pausing media, how can we mock repository to stub the values of `play`,`pause` audio & video
> > 
> > 
> > Mock the viewModel not helpers.
> 
> I have mocked the viewModel itself and not using helper classes in this test case but I didn't deleted it yet as when we have to write test cases for helper classes and we need them. So, can i delete it for now or leave it as it is?

Add it then. When its needed.

**Review by kjawadDeveloper2 — APPROVED** [review]

**Review by Amean — APPROVED** [review]

**Pull request merged**